src/utils/typedData.ts

```diff
@@ -107,130 +152,183 @@ function getMerkleTreeType(types: TypedData['types'], ctx: Context) {
+        // TODO: should this be skipped for v5 and added as ByteArray in v6?
+        const elements = splitLongString(data as string).map(getHex);
+        return [type, revisionConfiguration[revision].hashMethod(elements)];
+      }
+      return [type, getHex(data as string)];
+    }
+    default: {
+      return [type, getHex(data as string)];
+    }
+  }
+}
 
 /**
  * Encode the data to an ABI encoded Buffer. The data should be a key -> value object with all the required values.
  * All dependent types are automatically encoded.
  */
-export const encodeData = <T extends TypedData>(
+export function encodeData<T extends TypedData>(
   types: T['types'],
   type: string,
-  data: T['message']
-) => {
-  const [returnTypes, values] = types[type].reduce<[string[], string[]]>(
+  data: T['message'],
+  revision: Revision = Revision.Legacy
+) {
+  const targetType = types[type] ?? presetTypes[type];
+  const [returnTypes, values] = targetType.reduce<[string[], string[]]>(
     ([ts, vs], field) => {
-      if (data[field.name] === undefined || data[field.name] === null) {
+      if (data[field.name] === undefined || (data[field.name] === null && field.type !== 'enum')) {
         throw new Error(`Cannot encode data: missing data for '${field.name}'`);
       }
 
       const value = data[field.name];
-      const [t, encodedValue] = encodeValue(types, field.type, value, {
-        parent: type,
-        key: field.name,
-      });
+      const ctx = { parent: type, key: field.name };
+      const [t, encodedValue] = encodeValue(types, field.type, value, ctx, revision);
 
       return [
         [...ts, t],
         [...vs, encodedValue],
       ];
     },
-    [['felt'], [getTypeHash(types, type)]]
+    [['felt'], [getTypeHash(types, type, revision)]]
   );
 
   return [returnTypes, values];
-};
+}
 
 /**
  * Get encoded data as a hash. The data should be a key -> value object with all the required values.
  * All dependent types are automatically encoded.
  */
-export const getStructHash = <T extends TypedData>(
+export function getStructHash<T extends TypedData>(
   types: T['types'],
   type: string,
-  data: T['message']
-) => {
-  return computeHashOnElements(encodeData(types, type, data)[1]);
-};
+  data: T['message'],
+  revision: Revision = Revision.Legacy
+) {
+  return revisionConfiguration[revision].hashMethod(encodeData(types, type, data, revision)[1]);
+}
 
 /**
- * Get the EIP-191 encoded message to sign, from the typedData object.
+ * Get the EIP-191|EIP-712 encoded message to sign, from the typedData object.
  */
-export const getMessageHash = (typedData: TypedData, account: BigNumberish): string => {
+export function getMessageHash(typedData: TypedData, account: BigNumberish): string {
   if (!validateTypedData(typedData)) {
     throw new Error('Typed data does not match JSON schema');
   }
 
+  const revision = identifyRevision(typedData) as Revision;
+  const { domain, hashMethod } = revisionConfiguration[revision];
```

Comment:
Consider asserting that `domain` is well defined and throw a proper error in case it isn't
i.e. it have all the required fields
Currently I think it will only throw a non indicative error in `encodeData` during reduce in case `domain` is undefined?

Reply:
If this PR is merged to v5 there will be a validation rework done for v6. 

It shouldn't be possible for `domain` to end up as `undefined` since `identifyRevision` is called within `validateTypedData`.